mountpoint-s3/src/inode.rs

```diff
@@ -822,7 +843,13 @@ impl SuperblockInner {
             InodeKindData::Directory { children, .. } => children.get(name),
         };
         match (remote, inode) {
-            (None, None) => Err(InodeError::FileDoesNotExist(name.to_owned(), parent.err())),
+            (None, None) => {
+                if self.config.cache_config.serve_lookup_from_cache {
+                    // Insert or update validity of negative cache entry.
+                    self.known_missing_keys.insert(parent.ino(), name);
```

Comment:
so since it's taking a global write lock for cache, is it still `try_update_fast_path`? I believe you've mentioned some experiments around the performance previously

Reply:
That's fair. It is a different lock, but the method names may be misleading. Let me think about it.